### PR TITLE
fix var names in aper phot, remove specutils version check

### PR DESCRIPTION
**Pull request by cshanahan1**
This PR changes some of the unit conversion related variable names in aperture photometry to make it clearer that the spectral y axis unit is not being used for the plugin's display unit, but rather the dataset unit type (flux or sb) along with the choice of flux and angle units from the UC plugin for display. This same change needs to be made in moment maps, but since much of the code where these variables are being referenced go away in #3156, that will be addressed there. This also removes a remnant of checking for old specutils version in moment maps which can go away now that the minimum version is 1.16

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -5,7 +5,8 @@ New Features
 ------------
 
 - Added flux/surface brightness translation and surface brightness
-  unit conversion in Cubeviz and Specviz. [#2781, #2940, #3088, #3111, #3113, #3129, #3139, #3149, #3155, #3178, #3185]
+  unit conversion in Cubeviz and Specviz. [#2781, #2940, #3088, #3111, #3113, #3129,
+                                           #3139, #3149, #3155, #3178, #3185, #3187]
```
Comment (pllim) [inline CHANGES.rst:9]: I don't think RST indentation is like Python indentation.

```suggestion
  #3139, #3149, #3155, #3178, #3185, #3187]
```

Comment (pllim) [reply CHANGES.rst:9]: When in doubt, use the "View file" option.

**Review by pllim — COMMENTED** [review]
Why does this look so familiar? Is this one of your accidentally reverted changes?

**Comment by rosteen** [comment]
What was the reason for using `display_unit` vs just reverting to the previous `display_flux_or_sb_unit`? I don't mind it, just curious.

**Comment by cshanahan1** [comment]
> What was the reason for using `display_unit` vs just reverting to the previous `display_flux_or_sb_unit`? I don't mind it, just curious.

My reasoning was that once all images in imviz are surface brightness, then the name would have to be changed, but if it is just 'display_unit' then it wouldn't have to change.

Kyle made the suggestion to eventually change it to 'sb_display_unit' though so that will happen, so if it has to change again anyway and you like flux_or_sb better i can do that.

**Comment by rosteen** [comment]
> > What was the reason for using `display_unit` vs just reverting to the previous `display_flux_or_sb_unit`? I don't mind it, just curious.
> 
> My reasoning was that once all images in imviz are surface brightness, then the name would have to be changed, but if it is just 'display_unit' then it wouldn't have to change.
> 
> Kyle made the suggestion to eventually change it to 'sb_display_unit' though so that will happen, so if it has to change again anyway and you like flux_or_sb better i can do that.

I think this is fine, thanks for the explanation.

**Review by rosteen — APPROVED** [review]

**Pull request merged**